hygieia-jenkins-plugin/src/main/java/jenkins/plugins/hygieia/HygieiaGlobalListener.java

```diff
@@ -34,58 +34,65 @@ public void onCompleted(Run run, @Nonnull TaskListener listener) {
+                        if (sonarResponse.getResponseCode() == HttpStatus.SC_CREATED) {
+                            listener.getLogger().println("Hygieia: Auto Published Sonar Data. " + sonarResponse.toString());
                         } else {
-                            listener.getLogger().println("Hygieia: Auto Published " + gcir.getToolName() + " Data. " + genericItemResponse.toString());
+                            listener.getLogger().println("Hygieia: Failed Auto Publishing Sonar Data. " + sonarResponse.toString());
                         }
+                    } else {
+                        listener.getLogger().println("Hygieia: Auto Published Sonar Result. Nothing to publish");
                     }
-                } catch (IOException e) {
-                    listener.getLogger().println("Hygieia: Error Auto Publishing Generic Collector Item data." + '\n' + e.getMessage());
+                } catch (IOException | URISyntaxException | ParseException e) {
+                    listener.getLogger().println("Hygieia: Error Auto Publishing Sonar data." + '\n' + e.getMessage());
                 }
+            }
 
+            if (!CollectionUtils.isEmpty(hygieiaGlobalListenerDescriptor.getHygieiaPublishGenericCollectorItems())) {
+                List<HygieiaPublisher.GenericCollectorItem> items = hygieiaGlobalListenerDescriptor.getHygieiaPublishGenericCollectorItems();
+                for (HygieiaPublisher.GenericCollectorItem item : items) {
+                    GenericCollectorItemBuilder genericCollectorItemBuilder = getGenericCollectorItemBuilder(buildResponse, run, hygieiaGlobalListenerDescriptor, item.toolName, item.pattern);
+                    try {
+                        List<GenericCollectorItemCreateRequest> genericCollectorItemCreateRequests = genericCollectorItemBuilder.getRequests();
+                        if (CollectionUtils.isEmpty(genericCollectorItemCreateRequests)) continue;
+                        for (GenericCollectorItemCreateRequest gcir : genericCollectorItemCreateRequests) {
+                            HygieiaResponse genericItemResponse = hygieiaService.publishGenericCollectorItemData(gcir);
+                            if (genericItemResponse.getResponseCode() == HttpStatus.SC_CREATED) {
+                                listener.getLogger().println("Hygieia: Auto Published " + gcir.getToolName() + " Data. " + genericItemResponse.toString());
+                            } else {
+                                listener.getLogger().println("Hygieia: Auto Published " + gcir.getToolName() + " Data. " + genericItemResponse.toString());
+                            }
+                        }
+                    } catch (IOException e) {
+                        listener.getLogger().println("Hygieia: Error Auto Publishing Generic Collector Item data." + '\n' + e.getMessage());
+                    }
+
+                }
```

Comment:
Anything changed here or the diff is just due to format change?

Reply:
just the format change, not sure why intellij did it that way.